### ISSUE TITLE
Banned Symbol Analyzers reports duplicate violations

### ISSUE THREAD

**Issue by kaylumah** [dotnet/roslyn-analyzers#5616]
### Analyzer

Diagnostic ID: [RS0030](https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.BannedApiAnalyzers/Core/SymbolIsBannedAnalyzer.cs)

Microsoft.CodeAnalysis.BannedApiAnalyzers 3.3.2

### Describe the bug

Banned symbols are reported twice.

### Steps To Reproduce
I have the following BannedSymbols.txt

```
T:System.DateTime;Always use System.DateTimeOffset over System.DateTime
P:System.DateTimeOffset.Now;Use ISystemClock.Now instead
```

The following code

```
public class TuesdayDiscountedPriceCalculator : IDiscountedPriceCalculator
    {
        public decimal GetDiscountedPrice(decimal originalPrice)
        {
            if (DateTime.Now.DayOfWeek == DayOfWeek.Tuesday)
            {
                return originalPrice / 2;
            }
            else
            {
                return originalPrice;
            }
        }
    }
```

### Expected behavior

One error is reported on line 9

### Actual behavior

The error is reported twice

## Additional context

If I change DateTime to DateTimeOffset the warning about not using Now is only shown once
